cached.js

```diff
@@ -0,0 +1,68 @@
+"use strict";
+
+var shouldupdate = require('./shouldupdate');
+
+/**
+ * Directly fetch `cache` to use outside of Omniscient.
+ * You can do this if you want to define functions that caches computed
+ * result to avoid recomputing if invoked with equal arguments as last time.
+ *
+ * @param {Function} Function that does a computation.
+ *
+ * @module cached
+ * @returns {Function}
+ * @api public
+ */
+module.exports = factory();
+
+/**
+ * Create a “local” instance of the `cache` with overriden defaults.
+ *
+ * ### Options
+ * ```js
+ * {
+ *   isEqualProps: function (currentProps, nextProps), // check props
+ * }
+ * ```
+ *
+ * @param {Object} [Options] Options with defaults to override
+ *
+ * @module cached.withDefaults
+ * @returns {Function} cached with overriden defaults
+ * @api public
+ */
+module.exports.withDefaults = factory;
+
+function factory (methods) {
+  var isEqual = (methods && methods.isEqualProps) || shouldupdate.isEqualProps;
+
+  /**
+   * Returns optimized version of given `f` function for repeated
+   * calls with an equal inputs. Returned function caches last input
+   * and a result of the computation for it, which is handy for
+   * optimizing `render` when computations are run on unchanged parts
+   * of state. Although note that only last result is cached so it is
+   * not practical to call it mulitple times with in the same `render`
+   * call.
+   *
+   * @param {Function} function doing computation
+   *
+   * @module shouldComponentUpdate.isEqualState
+   * @returns {Function} Optimized function.
+   * @api public
+   */
+  return function cached (f) {
```

Comment:
This will imply that you need to declare the cached function outside render, and not create a new cache every render right?